kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/ExpressionVisitor.java

```diff
@@ -105,22 +107,28 @@ private R visitScalarExpression(ScalarExpression expression) {
                 return visitIsNull(new Predicate(name, children));
             case "COALESCE":
                 return visitCoalesce(expression);
+            case "LIKE":
+                return visitLike(new Predicate(name, children));
             default:
                 throw new UnsupportedOperationException(
                     String.format("Scalar expression `%s` is not supported.", name));
         }
     }
 
     private static Predicate elemAsPredicate(List<Expression> expressions, int index) {
-        if (expressions.size() <= index) {
-            throw new RuntimeException(
-                String.format("Trying to access invalid entry (%d) in list %s", index,
-                    expressions.stream().map(Object::toString).collect(joining(","))));
-        }
-        Expression elemExpression = expressions.get(index);
+        Expression elemExpression = elemAsExpression(expressions, index);
         if (!(elemExpression instanceof Predicate)) {
             throw new RuntimeException("Expected a predicate, but got " + elemExpression);
         }
         return (Predicate) expressions.get(index);
     }
+
+    private static Expression elemAsExpression(List<Expression> expressions, int index) {
+        if (expressions.size() <= index) {
```

Comment:
is this change needed?

Reply:
no. with the recent changes this is no longer needed.